src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/components/table/table.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiTableOfRecords,
+  EuiFieldText,
+  keyCodes,
+} from '@elastic/eui';
+
+export class Table extends Component {
+  static propTypes = {
+    indexPattern: PropTypes.object.isRequired,
+    model: PropTypes.shape({
+      data: PropTypes.shape({
+        records: PropTypes.array.isRequired,
+        totalRecordCount: PropTypes.number.isRequired,
+      }).isRequired,
+      criteria: PropTypes.shape({
+        page: PropTypes.shape({
+          index: PropTypes.number.isRequired,
+          size: PropTypes.number.isRequired,
+        }).isRequired,
+        sort: PropTypes.shape({
+          field: PropTypes.string.isRequired,
+          direction: PropTypes.string.isRequired,
+        }).isRequired,
+      }).isRequired,
+    }),
+    deleteFilter: PropTypes.func.isRequired,
+    onDataCriteriaChange: PropTypes.func.isRequired,
+    fieldWildcardMatcher: PropTypes.func.isRequired,
+    saveFilter: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingFilterId: null,
+      editingFilterValue: null,
+    };
+  }
+
+  startEditingFilter = (id, value) => this.setState({ editingFilterId: id, editingFilterValue: value })
+  stopEditingFilter = () => this.setState({ editingFilterId: null })
+  onEditingFilterChange = e => this.setState({ editingFilterValue: e.target.value })
+
+  onEditFieldKeyPressed = ({ charCode }) => {
+    if (keyCodes.ENTER === charCode) {
```

Comment:
Can we handle the `ESC` key too, and call `stopEditingFilter` when it's pressed?